Server/VikingVault.Services/PaymentTogglingService.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using VikingVault.DataAccess;
+using VikingVault.Services.Abstractions;
+
+namespace VikingVault.Services
+{
+    public class PaymentTogglingService : IPaymentTogglingService
+    {
+        private VikingVaultDbContext _context;
+
+        public PaymentTogglingService(VikingVaultDbContext context)
+        {
+            _context = context;
+        }
+
+        public void ChangePaymentState(int id, bool value)
+        {
+            var automaticPayment = _context.AutomaticPayments.Find(id);
+            automaticPayment.IsEnabled = !value;
+            _context.Update(automaticPayment);
+            _context.SaveChanges();
+        }
+
+        public bool? IsPaymentEnabled(int id)
+        {
+            try
+            {
+                return _context.AutomaticPayments.Find(id).IsEnabled;
+            }
+            catch (Exception e)
+            {
+                return null;
```

Comment:
I noticed that in other methods in the application, in case of an exception occurs, you throw a custom exception and handle it in the controller. You should do the same here, to be consistent